btree/btree.c

```diff
@@ -3504,7 +3507,8 @@ int64_t btree_destroy_tick(struct m0_sm_op *smop)
 		if (bop->bo_i == NULL)
 			return M0_ERR(-ENOMEM);
 
-		tree_put(bop->bo_arbor->t_desc);
+		if(bop->bo_arbor->t_desc->t_ref > 0)
+			tree_put(bop->bo_arbor->t_desc);
```

Comment:
There is no need to do this check of t_ref. Just call tree_put and let him handle this.

Reply:
This is basically to avoid assert in tree_put() as tree_put() currently doesnt handle the case where the t_ref is zero.
This case mostly occurs in the basic_tree_op UT.

Reply:
If that is the case then the call to destroy_tree_tick() with (t_ref ==0) is invalid too. You want to put an assert to check for this in the current routine.

Reply:
I have modified the logic to gracefully handle this condition.
This change does not remove the above line but instead returns an error code when t_ref==0 case is hit. As a result, the memory for bop->bo_arbor is freed properly.